data/battle_scripts_1.s

```diff
@@ -8663,14 +8685,15 @@ BattleScript_ShedSkinActivates::
 	end3
 
 BattleScript_ActivateWeatherAbilities:
-	copybyte sBATTLER, gBattlerAttacker
-	setbyte gBattlerAttacker, 0
+	savetarget
+	setbyte gBattlerTarget, 0
 BattleScript_ActivateWeatherAbilities_Loop:
-	activateweatherchangeabilities BS_ATTACKER
+	copybyte sBATTLER, gBattlerTarget
+	activateweatherchangeabilities BS_TARGET
 BattleScript_ActivateWeatherAbilities_Increment:
-	addbyte gBattlerAttacker, 1
-	jumpifbytenotequal gBattlerAttacker, gBattlersCount, BattleScript_ActivateWeatherAbilities_Loop
-	copybyte gBattlerAttacker, sBATTLER
+	addbyte gBattlerTarget, 1
+	jumpifbytenotequal gBattlerTarget, gBattlersCount, BattleScript_ActivateWeatherAbilities_Loop
+	restoretarget
```

Comment:
This can potentially break other abilities that trigger on weather change (Protosynthesis and Ice Face).
Ice face has its own check to prevent its trigger on more than 1 weather activation, but Protosynthesis doesn't, so it might trigger on each Sun turn

Reply:
Protosynthesis has a check to `gSpecialStatuses[battlerId].weatherAbilityDone` that isn't reset from turn to turn and is only used for Protosynthesis.